### PR TITLE
[Subscribing] Blank display name

### PR DESCRIPTION
**Pull request by Claire1817**
## Type of change

- [ ] Feature
- [X] Bugfix
- [ ] Technical
- [ ] Other :

## Content
Add a call to getProfile in the defaultSyncTask and store the result in the DB.

## Motivation and context
#5497 

## Screenshots / GIFs
Screenshot_1647875995

## Tests

- Create an account
- Open the left panel and check if there is the display name

## Tested devices

- [ ] Physical
- [X] Emulator
- OS version(s): API30

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [X] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [X] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncTask.kt [hunk 1]
```diff
@@ -104,7 +106,11 @@ internal class DefaultSyncTask @Inject constructor(
         val isInitialSync = token == null
         if (isInitialSync) {
             // We might want to get the user information in parallel too
-            userStore.createOrUpdate(userId)
+            val user = tryOrNull { session.getProfile(userId) }
+            userStore.createOrUpdate(
+                    userId = userId,
+                    displayName = user?.get(ProfileService.DISPLAY_NAME_KEY) as? String,
```
[the remark this answers is not in the document]
Comment (ouchadam) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncTask.kt:112]: are there other places we're using this logic, could be a function in the class for now until we have another usage?

or do you mean helpers for `fun User.getDisplayName() : String?`

[the remark this answers is not in the document]
Comment (Claire1817) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncTask.kt:112]: @ganfra if we add the small part in a private method in SyncTask class is enough for you ?

[the remark this answers is not in the document]
Comment (Florian14) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncTask.kt:112]: I noticed a similar code in that [file](https://github.com/vector-im/element-android/blob/3f39c5dae1cf4e9d33190a1fc421befb650d02b2/vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt#L244-L249) 
We can probably have a function returning a User object from the given userId and the related profile dictionary?

[the remark this answers is not in the document]
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncTask.kt:112]: Yes, @Claire1817 add it to the interface ProfileService please.

Something like:

```kotlin
    suspend fun getProfileAsUser(userId: String): User {
        return getProfile(userId).let { dict ->
            User(
                    userId = userId,
                    displayName = dict[DISPLAY_NAME_KEY] as? String,
                    avatarUrl = dict[AVATAR_URL_KEY] as? String
            )
        }
    }
```

File: vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt [hunk 2]
```diff
@@ -241,11 +241,11 @@ class UserListViewModel @AssistedInject constructor(
...
                         .sortedBy { it.toMatrixItem().firstLetterOfDisplayName() }
                 val userProfile = if (MatrixPatterns.isUserId(search)) {
-                    val json = tryOrNull { session.getProfile(search) }
+                    val user = tryOrNull { session.getProfileAsUser(search) }
                     User(
                             userId = search,
-                            displayName = json?.get(ProfileService.DISPLAY_NAME_KEY) as? String,
-                            avatarUrl = json?.get(ProfileService.AVATAR_URL_KEY) as? String
+                            displayName = user?.displayName,
+                            avatarUrl = user?.avatarUrl
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt:248]: You don't have to create a new User object if you already have one, just handle the nullability case:
`tryOrNull { session.getProfileAsUser(search) } ?: User(search)`

Comment (Claire1817) [reply vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt:248]: nice catch !

File: vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt [hunk 3]
```diff
@@ -214,14 +214,10 @@ class UserListViewModel @AssistedInject constructor(
                 ThreePidUser(email = search, user = null)
             } else {
                 try {
-                    val user = session.getProfileAsUser(foundThreePid.matrixId)
+                    val user = tryOrNull { session.getProfileAsUser(search) } ?: User(search)
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt:216]: I didn't look at the whole code, `search` is for an email, you should pass `foundThreePid.matrixId`, as before

**Review by Florian14 — APPROVED** [review]
LGTM, I let someone from the core team approve your changes

**Review by ouchadam — APPROVED** [review]
LGTM, thanks for the updates! :100:

**Pull request merged**